src/looprestoration.rs

```diff
@@ -372,64 +376,49 @@ unsafe fn wiener_rust<BD: BitDepth>(
         edges,
     );
 
-    let mut hor: [uint16_t; 27300] = [0; 27300];
-    let mut hor_ptr: *mut uint16_t = hor.as_mut_ptr();
-    let filter: *const [int16_t; 8] = ((*params).filter.0).as_ptr();
+    // Values stored between horizontal and vertical filtering don't
+    // fit in a uint8_t.
+    let mut hor = [0u16; 70 /*(64 + 3 + 3)*/ * REST_UNIT_STRIDE];
+
+    let filter = &params.filter.0;
     let bitdepth = bd.bitdepth().as_::<libc::c_int>();
-    let round_bits_h = 3 as libc::c_int + (bitdepth == 12) as libc::c_int * 2;
-    let rounding_off_h = (1 as libc::c_int) << round_bits_h - 1;
-    let clip_limit = (1 as libc::c_int) << bitdepth + 1 + 7 - round_bits_h;
-    let mut j = 0;
-    while j < h + 6 {
-        let mut i = 0;
-        while i < w {
-            let mut sum = (1 as libc::c_int) << bitdepth + 6;
+    let round_bits_h = 3 + (bitdepth == 12) as libc::c_int * 2;
+    let rounding_off_h = 1 << round_bits_h - 1;
+    let clip_limit = 1 << bitdepth + 1 + 7 - round_bits_h;
+    let chunks = tmp
+        .chunks_exact(REST_UNIT_STRIDE)
+        .zip(hor.chunks_exact_mut(REST_UNIT_STRIDE));
+    for (_, (tmp, hor)) in (0..h + 6).zip(chunks) {
```

Comment:
A slice plus enumerate would be better.  Simpler, more efficient implementation I'm pretty sure.

Reply:
Sorry it's hard to type on my phone lol.  `chunks[..h + 6].iter().enumerate()`.

Reply:
This requires `as_chunks` right? I don't want to use unstable features so I can't use that approach for now. Is this something we can cleanup later?

Reply:
Oh I misread the iterator code.  I thought `chunks` was a slice.  In that case you can do `chunks.iter().take(h + 6).enumerate()`.

Reply:
Done!